### ISSUE TITLE
Replication slot can grow without bound if there are writes to other databases on the same server

### ISSUE THREAD

**Issue by ethan-readyset** [readysettech/readyset#263]
### Summary

If there are databases other than the one we're connected to on the same Postgres server that are receiving steady writes but the database we're connected to is **not** receiving writes, the replication slot can grow without bound.

### Description

When we start up the Postgres connector, we create a publication for all the tables in a given database and then create a replication slot using that publication. This means that we only receive replication events for the tables in the database we're connected to. However, there may be other databases on the database server that are receiving writes, resulting in WAL entries being written. If the database ReadySet is connected to is not receiving any writes but the other databases are, the WAL will continue to grow, but we will continue to report our position as the last COMMIT LSN we wrote to our base tables. This means that our replication slot will grow further and further behind as the other databases receive writes.

### Expected behavior

If the database we're connected to isn't receiving writes but other databases on the Postgres server are receiving writes, we should correctly ACK our position in the WAL in response to keepalive messages to ensure that our replication slot does not grow without bound.

### Actual behavior

If the database we're connected to isn't receiving writes but other databases on the Postgres server are receiving writes, we currently respond to keepalive messages with the LSN of the last COMMIT we processed, which remains constant as long as the database we're connected to isn't receiving any writes.

### Steps to reproduce 

* Create two databases on the same Postgres server
* Start ReadySet connected to one of the databases
* Perform steady writes to the other database
* Observe that ReadySet's replication slot size grows without bound via `SELECT slot_name, pg_size_pretty(pg_wal_lsn_diff(pg_current_wal_lsn(), restart_lsn)) AS slot_size FROM pg_replication_slots`

### ReadySet version

```
noria=> SHOW READYSET VERSION;
      ReadySet      |             Version Information
--------------------+---------------------------------------------
 release version    | unknown-release-version
 commit id          | 1b7efc9ff93e1579a0ebc585d3c3c884fc19b7dc
 platform           | aarch64-apple-darwin
 rustc version      | rustc 1.70.0-nightly (f63ccaf25 2023-03-06)
 profile            | debug
 optimization level | 0
```

### Upstream DB type and version

```
noria=# SELECT VERSION();
                                                              version
-----------------------------------------------------------------------------------------------------------------------------------
 PostgreSQL 14.6 (Debian 14.6-1.pgdg110+1) on aarch64-unknown-linux-gnu, compiled by gcc (Debian 10.2.1-6) 10.2.1 20210110, 64-bit
(1 row)
```

### Instance Details

N/A

### Deployment Details

Custom-built binary

macOS 13.4.1 (c) (22F770820d)

### Logs

N/A

**Comment by ethan-readyset** [comment]
I think the solution here is as follows. When we receive a keepalive request from Postgres and we are not in the middle of a transaction (i.e. the last event we processed was a COMMIT or a keepalive request), we should respond to the keepalive request with the LSN given to us in the keepalive request.

**Comment by gvsg-rs** [comment]
@ethan-readyset Is this a dupe of [REA-2624](https://linear.app/readyset/issue/REA-2624/prevent-building-up-transaction-logs-in-aurora-when-writes-are-not)?

**Comment by ethan-readyset** [comment]
@gvsg-rs Yup — the issue described in [REA-2624](https://linear.app/readyset/issue/REA-2624/prevent-building-up-transaction-logs-in-aurora-when-writes-are-not) would happen if there were writes to *any* other databases on the Postgres server, so it's not specific to Aurora
